### ISSUE TITLE
Tutorial part 12

### ISSUE THREAD

**Issue by douglascrp** [Alfresco/Aikau#1107]
Following the tutorial, at the part 12, I have found 2 new issues.

https://github.com/Alfresco/Aikau/blob/master/tutorial/chapters/Tutorial12.md#step-3---sort-direction-toggle
When I use the menu items, the list is sorted ok, but the menu items do not reflect the sort state when I use the column headers.

https://github.com/Alfresco/Aikau/blob/master/tutorial/chapters/Tutorial12.md#step-4---multiple-item-selection
For the widget **alfresco/documentlibrary/AlfSelectDocumentListItems**, the actions **Invert**, **Folders** and **Documents** do nothing.
The actions **All** and **None** work perfectly, and the widget also reflects the list's selection state when I select or de-select all, some or no item in the list.

**Comment by AFaust** [comment]
For the **Invert** action we have recently discovered that the alfresco/renderers/Selector is missing the ```onClick``` function that the alfresco/lists/ItemSelectionMixin expects. Incidentally, the mixin itself provides a ```onSelectionClick``` which the Selector inherits..
@douglascrp: For **Folders** and **Documents** it is imperative that the currentItem of each list entry has a very specific object structure which either the alfresco/renderers/_JsNodeMixin or alfresco/lists/views/layouts/_MultiItemRendererMixin prepares IF the currentItem has a *node* property. This is also handled by the alfresco/lists/ItemSelectionMixin.

**Comment by draperd** [comment]
Thanks for raising these concerns - I'll dig into them as soon as I can. Keep them coming !

**Comment by draperd** [comment]
@douglascrp Can you describe what you mean by "the menu items do not reflect the sort state" - are you saying that when you sort by clicking on the column headers that the menu option doesn't change?

I can't remember if it was ever the intention in the tutorial for this to be demonstrated, but it should be possible. I can't find any unit test covering this scenario and it should be possible (and ideally happening automatically when defaults are used).

We don't appear to have any unit tests to cover this so we should probably investigate and add some. Also, it's worth bearing in mind that the options in the AlfSelectDocumentListItems are just defaults provided for use with the Document Library - it is possible to override them with a custom list. As @AFaust points out there are issues but we can't really change the defaults at this point although we could look to try and make those behaviours more generic (you'd hope that invert would work for example - folders and document are obviously much more context sensitive).

**Comment by draperd** [comment]
I've raised https://issues.alfresco.com/jira/browse/AKU-996 to address this.

**Comment by douglascrp** [comment]
@draperd 
About "are you saying that when you sort by clicking on the column headers that the menu option doesn't change?". Yes, that's what what I meant.

**Comment by draperd** [comment]
OK... I've started looking into this, I will update the tutorial... but for reference - with regards to the `alfresco/menus/AlfMenuBarToggle` widget you should be able to use this configuration:

```JAVASCRIPT
{
   id: "SORT_TOGGLE",
   name: "alfresco/menus/AlfMenuBarToggle",
   config: {
      subscriptionTopic: "ALF_DOCLIST_SORT",
      subscriptionAttribute: "direction",
      checked: true,
      checkedValue: "ascending",
      onConfig: {
         title: "Change sort order to descending",
         iconClass: "alf-sort-ascending-icon",
         iconAltText: "Sorted ascending",
         publishTopic: "ALF_DOCLIST_SORT",
         publishPayload: {
            direction: "ascending"
         }
      },
      offConfig: {
         title: "Change sort order to ascending",
         iconClass: "alf-sort-descending-icon",
         iconAltText: "Sorted descending",
         publishTopic: "ALF_DOCLIST_SORT",
         publishPayload: {
            direction: "descending"
         }
      }
   }
}
```

The key changes are setting:
- subscriptionTopic
- subscriptionAttribute
- checkedValue

The `alfresco/menus/AlfMenuBarToggle` is not specific to sorting so we should not change the defaults (and cannot for backwards compatibility reasons).

I'll look into the drop-down menu next...

**Comment by draperd** [comment]
OK... so some configuration is also required for the `alfresco/menus/AlfMenuBarSelect` - but also some code changes are required to HeaderCell and AlfSortablePaginatedList, I'll get these fixed this sprint and update the tutorial

**Comment by draperd** [comment]
This is now resolved in the 1.0.73 release that is now available

**Comment by draperd** [comment]
Bit late, but I realised that we'd missed the invert fix - there's a PR for it ( #1146 ) which will be available in 1.0.77
